cmd/inferTime.go

```diff
@@ -0,0 +1,163 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+/*
+Copyright © 2020 Jean-Marc Meessen, ON4KJM <on4kjm@gmail.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//InferTimeBlock contains the information describing a time gap
+type InferTimeBlock struct {
+	lastRecordedTime time.Time
+	nextValidTime    time.Time
+
+	//Number of records without actual time
+	noTimeCount int
+
+	//First log entry with missing date
+	logFilePosition int
+
+	//Computed time interval (in seconds)
+	deltatime int
+}
+
+//displayTimeGapInfo will print the details stored in an InferTimeBlock
+func (tb *InferTimeBlock) displayTimeGapInfo() string {
+	timeFormat := "2006-01-02 15:04"
+	buffer := ""
+	buffer = buffer + fmt.Sprintf("Last Recorded Time:                 %s\n", tb.lastRecordedTime.Format(timeFormat))
+	buffer = buffer + fmt.Sprintf("next Recorded Time:                 %s\n", tb.nextValidTime.Format(timeFormat))
+	buffer = buffer + fmt.Sprintf("Log position of last recorded time: %d\n", tb.logFilePosition)
+	buffer = buffer + fmt.Sprintf("Nbr of entries without time:        %d\n", tb.noTimeCount)
+	buffer = buffer + fmt.Sprintf("Computed interval:                  %d\n", tb.deltatime)
+
+	return buffer
+}
+
+//finalizeTimeGap makes the necessary checks and computation
+func (tb *InferTimeBlock) finalizeTimeGap() error {
+	var err error
+	err = nil
+
+	//Check that lastRecordedTime and nextValidTime are not null
+	nullTime := time.Time{}
+	if tb.lastRecordedTime == nullTime {
+		errMsg := "Fatal error: gap start time is empty"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+	if tb.nextValidTime == nullTime {
+		errMsg := "Fatal error: gap end time is empty"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+
+	//Are the two times equal?
+	if tb.nextValidTime == tb.lastRecordedTime {
+		errMsg := "Fatal error: the start and end gap times are equal"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
```

Comment:
```suggestion
		return errors.New("start and end gap times are equal")
```

Few remarks here:
- When you don't need to format your errors message (or wrap errors), it is recommended to use `errors.New` instead of `fmt.Errorf`. `fmt.Errorf` is "kinda" equivalent to `errors.New(fmt.Sprintf("Formated %v error", value)))`. `fmt.Errorf` also handles error wrapping but we're not there yet :). In your case you don't need the formatting sooo, let's  skip to `errors.New`  directly :)
- I removed the `Fatal error` because I believe it's the caller responsibility to actually decide if this error is fatal or not. You might want to apply some retries on this logic, thus the `fatal` error would be only decided after all the retries fails... 

For example here's how could look the call site :) 

```
var tb InferTimeBlock
// ....
if err := tb.finalizeTimeGap(); err != nil {
       log.Println("Fatal error:", err)
       os.Exit(1) // and so on...
}
```

Reply:
Implemented